File: 0455-assign-cookies/0455-assign-cookies.py
class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        g.sort()
        s.sort()
        
        i = 0  # pointer for g
        j = 0  # pointer for s
        
        while i < len(g):
            while j < len(s) and  g[i] > s[j]:
                j += 1
            
            if j < len(s):
                i += 1
                j += 1
            else:
                break
        
        return i
    
# Sorting and Two Pointers
# Time Complexity: O(nlogn + mlogm)
# Space Complexity :O(1)
                
                
            

File: 0455-assign-cookies/0455-assign-cookies.java
class Solution {
    public int findContentChildren(int[] g, int[] s) {
        Arrays.sort(g);
        Arrays.sort(s);
        
        int i = 0; // pointer for g
        int j = 0; // pointer for s
        
        while(i < g.length){
            while(j < s.length && g[i] > s[j]){
                j++;
            }
            if(j < s.length){
                i++;
                j++;
            }
            else{
                break;
            }
        }
        
        return i;        
            
    }
}


// Sorting and Two Pointers
// Time Complexity: O(nlogn + mlogm)
// Space Complexity :O(1)
                